datasources/gab/search_gab.py

```diff
@@ -0,0 +1,83 @@
+"""
+Import scraped Gab data
+"""
+import datetime
+import re
+
+from backend.lib.search import Search
+
+
+class SearchGab(Search):
+    """
+    Import scraped gab data
+    """
+    type = "gab-search"  # job ID
+    category = "Search"  # category
+    title = "Import scraped Gab data"  # title displayed in UI
+    description = "Import Gab data collected with an external tool such as Zeeschuimer."  # description displayed in UI
+    extension = "ndjson"  # extension of result file, used internally and in UI
+    is_from_extension = True
+    fake = ""
+
+    # not available as a processor for existing datasets
+    accepts = [None]
+
+    def get_items(self, query):
+        """
+        Run custom search
+
+        Not available for Gab
+        """
+        raise NotImplementedError("Gab datasets can only be created by importing data from elsewhere")
+
+    @staticmethod
+    def map_item(post):
+        """
+        Parse Gab post
+
+        :param node:  Data as received from Gab
+        :return dict:  Mapped item
+        """
+        
+        post_time = datetime.datetime.strptime(post["ca"], "%Y-%m-%dT%H:%M:%S.%fZ")
+        mapped_item = {
+            "id": post["i"],
+            "created_at": post["ca"],
+            "body": post["c"],
+            "url": post["ul"],
+            "reaction_count": post["fc"] if post["fc"] else 0,
+            "reposts_count": post["rbc"],
+            "replies_count": post["rc"],
+            "group_id": post["g"]["id"] if "g" in post else None,
+            "group_title": post["g"]["title"] if "g" in post else None,
+            "group_description": post["g"]["description"] if "g" in post else None,
+            "group_member_count": post["g"]["member_count"] if "g" in post else None,
+            "group_is_private": post["g"]["is_private"] if "g" in post else None,
+            "group_member_count": post["g"]["url"] if "g" in post else None,
```

Comment:
This line appears duplicative with `group_url` below.

Reply:
whoops thanks